packages/http-client/src/index.ts

```diff
@@ -1,80 +1,129 @@
 import fetch from "isomorphic-fetch";
-import type { Includes, Project, Page } from "./index.d";
+import type { Project } from "./index.d";
 
 export const loadProject = async ({
   apiUrl,
   projectId,
-  include = { tree: true, props: true, breakpoints: true, pages: false },
 }: {
   apiUrl: string;
   projectId: string;
-  include?: Includes<boolean>;
-}): Promise<Project> => {
-  if (apiUrl === undefined) {
-    throw new Error("Webstudio API URL is required.");
-  }
-  const urls = [];
-  const pages: Record<string, unknown> | null = {};
-
-  const baseUrl = new URL(apiUrl);
-  const treeUrl = new URL(`/rest/tree/${projectId}`, baseUrl);
-  const propsUrl = new URL(`/rest/props/${projectId}`, baseUrl);
-  const breakpointsUrl = new URL(`/rest/breakpoints/${projectId}`, baseUrl);
-  const pagesUrl = new URL(`/rest/pages/${projectId}`, baseUrl);
-
-  const [tree, props, breakpoints, listOfPages] = await Promise.all([
-    include.tree && (await fetch(treeUrl)).json(),
-    include.props && (await fetch(propsUrl)).json(),
-    include.breakpoints && (await fetch(breakpointsUrl)).json(),
-    include.pages && (await fetch(pagesUrl)).json(),
-  ]);
+}): Promise<Project | string> => {
+  try {
+    if (apiUrl === undefined) {
+      throw new Error("Webstudio API URL is required.");
+    }
+    const urls = [];
+    const gatheredPages: Project = {};
 
-  if (include.pages && listOfPages.pages && listOfPages.pages.length > 0) {
-    for (const page of listOfPages.pages) {
+    const baseUrl = new URL(apiUrl);
+    const pagesUrl = new URL(`/rest/pages/${projectId}`, baseUrl);
+    const listOfPages = await (await fetch(pagesUrl)).json();
+    if (listOfPages === undefined || listOfPages.errors) {
+      throw new Error(`Project ${projectId} needs to be published first`);
+    }
+    const { pages, homePage } = listOfPages;
+    if (homePage) {
+      const { id } = homePage;
+      const treeUrl = new URL(`/rest/tree/${projectId}/${id}`, baseUrl);
+      const propsUrl = new URL(`/rest/props/${projectId}/${id}`, baseUrl);
+      const breakpointsUrl = new URL(`/rest/breakpoints/${projectId}`, baseUrl);
+      const cssUrl = new URL(`/rest/css/${projectId}/${id}`, baseUrl);
       urls.push({
-        url: new URL(`/rest/tree/${projectId}/${page.id}`, baseUrl),
-        path: page.path,
+        page: { ...homePage },
```

Comment:
why do you spread instead of just using? Is it mutating it or something?
```suggestion
        page: homePage,
```